### PR TITLE
add `-show-url` option for `ui` command

### LINKED ISSUE #11193: Option to show url with `nomad ui -authenticate`
### Proposal

Option to show url with `nomad ui -authenticate` instead of opening it directly in the default browser (like `-show-url` or `-no-open`)

### Use-cases

- I use different browser profiles for different things
- with tmux sometimes it doesn't work (the cli just hangs)

### Attempted Solutions

open the web ui manually and copy/paste the token

### PR DESCRIPTION
**Pull request by yaroot**
closes #11193

### CODE REVIEW

File: command/ui.go [hunk 1]
```diff
@@ -184,7 +188,9 @@ func (c *UiCommand) Run(args []string) int {
 	} else {
 		c.Ui.Output(fmt.Sprintf("Opening URL %q", url.String()))
 	}
-	if err := open.Start(url.String()); err != nil {
+	if showUrl {
+		c.Ui.Output(fmt.Sprintf("URL for web UI: %s", url.String()))
```
Comment (lgfa29) [inline command/ui.go:192]: This message will cause the URL to be displayed twice:

```console
$ nomad ui -show-url
Opening URL "http://127.0.0.1:4646"
URL for web UI: http://127.0.0.1:4646
```

Perhaps we can delay outputting the message until all conditions are checked? So something like:

```go
var outptut string
if authenticate && ottSecret != "" {
	output = fmt.Sprintf("Opening URL %q with one-time token", url.String()))
	url.RawQuery = fmt.Sprintf("ott=%s", ottSecret)
} else {
	output = fmt.Sprintf("Opening URL %q", url.String()))
}

if showUrl {
	c.Ui.Output(fmt.Sprintf("URL for the Nomad Web UI: %s", url.String()))
	return 0
}

if err := open.Start(url.String()); err != nil {
	c.Ui.Error(fmt.Sprintf("Error opening URL: %s", err))
	return 1
}

c.Ui.Output(output)
return 0
```

What do you think?

**Review by lgfa29 — CHANGES_REQUESTED** [review]
Thank you for the PR @yaroot!

From my test, there's a duplicate message with the URL when the new flag is passed. I have a suggestion on how we can avoid it, but feel free to implement it another way if you find a better solution 🙂

File: command/ui.go (hunk 1 above)
Comment (yaroot) [reply command/ui.go:194]: Ah.. fixed

**Review by lgfa29 — APPROVED** [review]
Thank you for the fix @yaroot, things are working as expected now 🙂 

I pushed a couple of commits to enhance the documentation and also to add a CHANGELOG entry.

Thank you for the contribution!

**Pull request merged**